contract/models/contract.py

```diff
@@ -123,6 +123,41 @@ class ContractContract(models.Model):
         copy=False,
         track_visibility="onchange",
     )
+    modification_ids = fields.One2many(
+        comodel_name='contract.modification',
+        inverse_name='contract_id',
+        string='Modifications',
+    )
+
+    @api.multi
+    def write(self, vals):
+        res = super(ContractContract, self).write(vals)
+        if 'modification_ids' in vals:
```

Comment:
You should do this instead on create/write of the subobject. If not, no notification will be sent in case of direct creation through code or specific UI. We have to take into account multi-creation on specific code for sending all the modification in one batch.